training/creating_dataset.py

```diff
@@ -42,13 +83,149 @@ def is_path_correct(path: str) -> bool:
     """
     if not os.path.isfile(path):
         raise ValueError(f"Path {path} does not exist")
-        # return False
+
     if not (path.endswith(".db") or path.endswith(".json")):
         raise ValueError(f"Expected db or json file, got {path}")
-        # return False
+
     return True
 
 
+def split_characteristics_into_groups(
+    characteristics: list,
+) -> tuple[set[str], set[str], set[str], set[str], list[str]]:
+    """
+    Split given characteristics list into five categories according to sets and dictionaries with characteristics.
+    Categories depend on way of extracting data.
+    :param characteristics: list of characteristics
+    :return: sets and list of characteristics after split
+    """
+    speeds = OTHER_SPEEDS.intersection(characteristics)
+    weaknesses = WEAKNESSES.intersection(characteristics)
+    resistances = RESISTANCES.intersection(characteristics)
+    special_characteristics = SPECIAL_CHARACTERISTICS.intersection(characteristics)
+    characteristics_rename = [
+        ch for ch in characteristics if ch in CHARACTERISTICS_RENAME.keys()
+    ]
+
+    return (
+        speeds,
+        weaknesses,
+        resistances,
+        special_characteristics,
+        characteristics_rename,
+    )
+
+
+def get_characteristic_from_list(cell_value, characteristic_name: str) -> int:
+    """
+    Function used for pd.Series.apply()\n
+    Retrieve the value associated with a specific characteristic from a list of dictionaries.
+    :param cell_value: Value of cell from df. A list of dictionaries, where each dictionary represents a characteristic
+            with 'type' and 'value' keys.
+    :param characteristic_name: The characteristic name to search for.
+    :return: The value associated with the specified characteristic if found, or 0 if not found.
+    """
+    if not cell_value or np.all(pd.isnull(cell_value)):
+        return 0
+
+    result = [
+        x.get("value") for x in cell_value if x.get("type") == characteristic_name
+    ]
+    return 0 if len(result) == 0 else result[0]
```

Comment:
1. Nie ma gwarancji, że tam zawsze są `type` i `value`? W sensie tu musi być ten `.get()`?
2. Może czytelniej byłoby pętlą for?
```
for dictionary in cell_value:
    if dictionary.get("type") == characteristic_name:
        return dictionary.get("value")

# required characteristic was not found
return 0
```

Reply:
Chyba nie rozumiem pytania 1.
w przypadkach gdzie jest używana funkcja jest gwarancja (po tym wcześniejszym przefiltrowaniu), że pola `type` i `value` istnieją, ale typ chyba zawsze musimy sprawdzić, żeby wiedzieć, że bierzemy wartość np. `fly` a nie inną